pkg/controllers/nodeclaim/termination/suite_test.go

```diff
@@ -269,4 +271,61 @@ var _ = Describe("Termination", func() {
 			ExpectExists(ctx, env.Client, node)
 		}
 	})
+	It("should not annotate the node if the NodeClaim has no terminationGracePeriod", func() {
```

Comment:
IIUC, this test needs somewhere to trigger the nodeclaim deletion. If you reconcile the nodeclaim but there's no deletion timestamp, it won't do anything, which is why I think this test passes anyway.

Reply:
Thanks, good catch.